x-pack/solutions/observability/test/api_integration_deployment_agnostic/apis/streams/permissions.ts

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { DeploymentAgnosticFtrProviderContext } from '../../ftr_provider_context';
+import {
+  StreamsSupertestRepositoryClient,
+  createStreamsRepositoryAdminClient,
+  createStreamsRepositoryEditorClient,
+} from './helpers/repository_client';
+import { disableStreams, enableStreams, forkStream } from './helpers/requests';
+
+export default function ({ getService }: DeploymentAgnosticFtrProviderContext) {
+  const roleScopedSupertest = getService('roleScopedSupertest');
+  let adminApiClient: StreamsSupertestRepositoryClient;
+  let editorApiClient: StreamsSupertestRepositoryClient;
+
+  describe('Fails on missing permissions', () => {
+    before(async () => {
+      adminApiClient = await createStreamsRepositoryAdminClient(roleScopedSupertest);
+      editorApiClient = await createStreamsRepositoryEditorClient(roleScopedSupertest);
+      await enableStreams(adminApiClient);
+      await forkStream(adminApiClient, 'logs', {
+        stream: {
+          name: 'logs.forked',
+        },
+        if: { always: {} },
+      });
+    });
+
+    after(async () => {
+      await disableStreams(adminApiClient);
+    });
+
+    it('cannot fork a stream with editor permissions', async () => {
+      const response = await editorApiClient
+        .fetch('POST /api/streams/{name}/_fork 2023-10-31', {
+          params: {
+            path: { name: 'logs' },
+            body: {
+              stream: {
+                name: 'logs.forked2',
+              },
+              if: { always: {} },
+            },
+          },
+        })
+        .expect(403);
+
+      // make sure that the right permissions are validated
+      expect(response.body).to.eql({
+        statusCode: 403,
+        error: 'Forbidden',
+        message: 'User does not have sufficient permissions to execute these actions',
+        attributes: {
+          data: {
```

Comment:
I wonder if we should try to return these in the similar format that Elasticsearch does, which is usually a text string along the lines of "user doesn't have the required privileges for resources X, this is granted by privilege "manage_pipeline"? 
Just for consistency but maybe it doesn't matter.
Else we'll report back _all_ privileges (some marked as true), rather than the key ones which are missing, so signal/noise ratio is worse.

Reply:
I can imagine it helps to see the whole list that is required for the current request - so you don't change the role setup and now you are missing something else you didn't know you needed, if that makes sense.

Reply:
It's not trivial to calculate the "this is granted by xyz" part, I think we should stay out of that.